nuxeo-common/src/main/java/org/nuxeo/common/utils/ZipUtils.java

```diff
@@ -474,4 +474,20 @@ public static boolean hasEntry(URL url, String entryName) throws IOException {
         return hasEntry(url.openStream(), entryName);
     }
 
+    /**
+     * Checks if the content of the {@link InputStream} is a valid zip.
+     * The method does not close the stream.
+     * @param stream the {@link InputStream} to be validated
+     * @return true if the {@link InputStream} is a valid zip, false otherwise
+     */
+    public static boolean isValid(InputStream stream) {
+        try {
+            ZipInputStream zip = new ZipInputStream(stream);
+            ZipEntry entry = zip.getNextEntry();
+            return entry != null;
+        } catch (IOException e) {
+            return false;
+        }
```

Comment:
You must close the `ZipInputStream ` too:

       try (ZipInputStream zip = new ZipInputStream(stream)) {
            return zip.getNextEntry() != null;
        } catch (IOException e) {
            return false;
        }